models/src/test/java/io/sphere/sdk/reviews/ReviewDraftTest.java

```diff
@@ -0,0 +1,142 @@
+package io.sphere.sdk.reviews;
+
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by njacinto on 21/04/15.
+ */
+public class ReviewDraftTest {
+    static final ReviewDraft DEFAULT_REVIEW;
+
+
+    static {
+        DEFAULT_REVIEW = ReviewDraft.of(ReviewImplTest.DEFAULT_PRODUCT_ID, ReviewImplTest.DEFAULT_CUSTOMER_ID)
+                .withAuthorName(ReviewImplTest.DEFAULT_AUTHOR_NAME).withTitle(ReviewImplTest.DEFAULT_TITLE)
+                .withText(ReviewImplTest.DEFAULT_TEXT).withScore(ReviewImplTest.DEFAULT_SCORE);
+    }
+
+    @Test
+    public void testOf() throws Exception {
+        ReviewDraft review = ReviewDraft.of(ReviewImplTest.DEFAULT_PRODUCT_ID, ReviewImplTest.DEFAULT_CUSTOMER_ID);
+        assertEquals(ReviewImplTest.DEFAULT_PRODUCT_ID, review.getProductId());
+        assertEquals(ReviewImplTest.DEFAULT_CUSTOMER_ID, review.getCustomerId());
+    }
+
+    @Test
+    public void testGetProductId() throws Exception {
+        assertEquals(ReviewImplTest.DEFAULT_PRODUCT_ID, DEFAULT_REVIEW.getProductId());
+    }
+
+    @Test
+    public void testGetCustomerId() throws Exception {
+        assertEquals(ReviewImplTest.DEFAULT_CUSTOMER_ID, DEFAULT_REVIEW.getCustomerId());
+    }
+
+    @Test
+    public void testGetAuthorName() throws Exception {
+        assertEquals(ReviewImplTest.DEFAULT_AUTHOR_NAME, DEFAULT_REVIEW.getAuthorName().get());
+    }
+
+    @Test
+    public void testGetTitle() throws Exception {
+        assertEquals(ReviewImplTest.DEFAULT_TITLE, DEFAULT_REVIEW.getTitle().get());
+    }
+
+    @Test
+    public void testGetText() throws Exception {
+        assertEquals(ReviewImplTest.DEFAULT_TEXT, DEFAULT_REVIEW.getText().get());
+    }
+
+    @Test
+    public void testGetScore() throws Exception {
+        assertTrue(ReviewImplTest.isScoreInValidRange(DEFAULT_REVIEW.getScore().get(), ReviewImplTest.DEFAULT_SCORE));
+    }
+
+    @Test
+    public void testWithTitle() throws Exception {
+        ReviewDraft review = ReviewDraft.of(ReviewImplTest.DEFAULT_PRODUCT_ID, ReviewImplTest.DEFAULT_CUSTOMER_ID);
+        assertFalse(review.getTitle().isPresent());
+        ReviewDraft reviewWith = review.withTitle(ReviewImplTest.DEFAULT_TITLE);
+        assertFalse(review==reviewWith);
```

Comment:
Hello,

It’s not comparing objects. It’s comparing references.

Could you please send me the code guidelines?

Regards,

Nuno Jacinto

> On 22 Apr 2015, at 09:20, Hajo <notifications@github.com> wrote:
> 
> In models/src/test/java/io/sphere/sdk/reviews/ReviewDraftTest.java <https://github.com/sphereio/sphere-jvm-sdk/pull/367#discussion_r28849579>:
> 
> > +    @Test
> > +    public void testGetText() throws Exception {
> > +        assertEquals(ReviewImplTest.DEFAULT_TEXT, DEFAULT_REVIEW.getText().get());
> > +    }
> > +
> > +    @Test
> > +    public void testGetScore() throws Exception {
> > +        assertTrue(ReviewImplTest.isScoreInValidRange(DEFAULT_REVIEW.getScore().get(), ReviewImplTest.DEFAULT_SCORE));
> > +    }
> > +
> > +    @Test
> > +    public void testWithTitle() throws Exception {
> > +        ReviewDraft review = ReviewDraft.of(ReviewImplTest.DEFAULT_PRODUCT_ID, ReviewImplTest.DEFAULT_CUSTOMER_ID);
> > +        assertFalse(review.getTitle().isPresent());
> > +        ReviewDraft reviewWith = review.withTitle(ReviewImplTest.DEFAULT_TITLE);
> > +        assertFalse(review==reviewWith);
> What does the coding guideline says about spaces in this case?
> Shouldn't we use equal to compare objects?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/sphereio/sphere-jvm-sdk/pull/367/files#r28849579>.
> 


-- 


<http://berlin.apidays.io>

commercetools GmbH
SPHERE.IO <http://www.sphere.io>  *The E-Commerce Platform.*

Adams-Lehmann-Strasse 44, 80797 München

www.commercetools.com

Amtsgericht München, HRB 161496

Geschäftsführer: Denis Werner, Dirk Hörig, Dr. Robert Zores